src/audit/forbidden_uses.rs

```diff
@@ -72,41 +57,46 @@ impl Audit for ForbiddenUses {
                             .primary()
                             .with_keys(&["uses".into()])
                             .annotated("use of this action is forbidden"),
-                    )
-                    .build(step.workflow())?,
+                    ),
             );
         };
 
         Ok(findings)
     }
+}
+
+impl Audit for ForbiddenUses {
+    fn new(state: &AuditState<'_>) -> Result<Self, AuditLoadError>
+    where
+        Self: Sized,
+    {
+        let Some(config) = state
+            .config
+            .rule_config(Self::ident())
+            .context("invalid configuration")
+            .map_err(AuditLoadError::Fail)?
+        else {
+            return Err(AuditLoadError::Skip(anyhow!("audit not configured")));
+        };
+
+        Ok(Self { config })
+    }
+
+    fn audit_step<'w>(&self, step: &Step<'w>) -> anyhow::Result<Vec<Finding<'w>>> {
+        self.process_step(step)?
+            .into_iter()
+            .map(|f| f.build(step.workflow()))
```

Comment:
Ideally the `build` calls would also be done in the `process_step` method.

I tried to solve this by adding a `fn document(&self) -> &'s impl AsRef<yamlpath::Document>;` to the `StepCommon` trait (whose implementations then return `self.workflow()` respectively `self.action()`). But unfortunately I was running into lifetime issues then when trying to use that `document()` method.

If you want, try it yourself as well; maybe I am just missing Rust knowledge there.

Reply:
Ah yeah, good thought. I'll review these changes as-is, but agreed that `process_step` should be doing the `FindingBuilder -> Finding` transform internally.